### PR TITLE
Final DQM plot updates for the Totem T2 in the special run 25.6, to be ready for special TOTEM release testing around 12-13.6

### PR DESCRIPTION
**Pull request by oljemark**
#### PR description:

For the TOTEM special foreseen on 25.6.2023, the new T2 telescope will be used (and removed after the run) to measure inelastic rate and veto diffractive p+p collisions, when measuring the elastic p+p cross section.
This PR is for a special TOTEM release to be ready for testing start of next week (https://github.com/cms-sw/cmssw/pull/41859#issuecomment-1584370879).

After the merging of PR #41859 and submission of backport PR #41916 , T2 Digis are successfully saved in the event and used to produce Digi-based DQM plots in common reconstruction with the rest of the PPS detectors. 
In this PR, the DQM plots that use the T2Segmentation helper (removed due to T2 RecHit and TotemGeometry dependence causing relVal issues detailed in the previous PR's) were added back with a hardcoded "geometry" mapping from channel number to (X,Y) position using the attached mapping on slide 1 (agreed with @vavati & E. Bossini to use slide 1, not slide 3), see comments here: https://github.com/cms-sw/cmssw/compare/master...CTPPS:cmssw:fo_nt2_dqm_final#diff-c26415cb77d83b1dbe3bdea47ca041d0e35f6ab203364c54d92c4b143b11ce47R47

[T2-map-latest-2023jun11.pdf](https://github.com/cms-sw/cmssw/files/11715483/T2-map-latest-2023jun11.pdf)

Several other T2 plots were also adjusted as to the T2 Digi demands used.

This PR is the final one needed for testing a special 13_0_X release for TOTEM, or possibly to be backported to 13_1_X too, see https://github.com/cms-sw/cmssw/pull/41916#issuecomment-1585592026

#### PR validation:

After the inclusion of PR #41916 in master, there were no other dependent packages brought in by checkdeps , so I ran successfully a PPS workflow (1044) & the DQM test file totemt2_dqm_test_common_cfg.py , which produced the expected plots.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This will hopefully be backported very soon (during the coming week) to 13_0_X, and possibly later to 13_1_X, as noted above. I will submit a backport PR later with the commits from PR #41916 and the ones from this PR.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@oljemark please specify whether this PR has to be included ONLY for the special Totem run, or if it can remain in the release also for the "normal" pp runs

**Comment by vavati** [comment]
> @oljemark please specify whether this PR has to be included ONLY for the special Totem run, or if it can remain in the release also for the "normal" pp runs

T2-DQM is included only in a special sequence which is not executed in regular runs. At worse it creates empty plots. Hence whatever is easier for you: it can be included only for the special run or stay in.

**Comment by perrotta** [comment]
> > @oljemark please specify whether this PR has to be included ONLY for the special Totem run, or if it can remain in the release also for the "normal" pp runs
> 
> T2-DQM is included only in a special sequence which is not executed in regular runs. At worse it creates empty plots. Hence whatever is easier for you: it can be included only for the special run or stay in.

Thank you @vavati 
@emanueleusai please suggest how to proceed here. If the issue of the empty plot is not relevant, or (probably even better) if they can be enabled/disabled with a flag, keeping the related code in the cmssw release is the easiest way to maintain and resume this code when needed

**Comment by vavati** [comment]
> Thank you @vavati @emanueleusai please suggest how to proceed here. If the issue of the empty plot is not relevant, or (probably even better) if they can be enabled/disabled with a flag, keeping the related code in the cmssw release is the easiest way to maintain and resume this code when needed

FYI this is the sequence which is activated only in special runs:
https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/ctpps_dqm_sourceclient-live_cfg.py#LL94C35-L94C35

**Comment by perrotta** [comment]
urgent

**Comment by vavati** [comment]
In case:
https://cmsoms.cern.ch/cms/runs/report?cms_run=368805&cms_run_sequence=GLOBAL-RUN
has T2 emulated data + Totem strips. Good for tests.

**Comment by emanueleusai** [comment]
Personally, I'd rather have the sequence activated/deactivated with a flag rather than having empty plots, but if it's urgent for the special run on 25.6, I'm fine with keeping it like this and add a flag switch later on.

**Comment by emanueleusai** [comment]
+1

File: DQM/CTPPS/plugins/TotemT2DQMSource.cc [hunk 1]
```diff
@@ -305,6 +352,14 @@ void TotemT2DQMSource::fillEdges(const TotemT2Digi& digi, const TotemT2DetId& de
   channelPlots_[detid].trailingEdgeCh->Fill(T2_BIN_WIDTH_NS_ * digi.trailingEdge());
 }
 
+void TotemT2DQMSource::fillBX(const TotemT2Digi& digi, const TotemT2DetId& detid, const int bx) {
+  const TotemT2DetId secId(detid.armId());
+  if (digi.hasLE()) {
```
Comment (perrotta) [inline DQM/CTPPS/plugins/TotemT2DQMSource.cc:357]:
```suggestion
  if (digi.hasLE()) {
    const TotemT2DetId secId(detid.armId());
```

**Review by perrotta — COMMENTED** [review]
Just a suggestion for a possible follow up PR

**Comment by perrotta** [comment]
> Personally, I'd rather have the sequence activated/deactivated with a flag rather than having empty plots, but if it's urgent for the special run on 25.6, I'm fine with keeping it like this and add a flag switch later on.

I agree.
@oljemark while we merge now this PR in master, in order to have it available for the special Totem run of June 25, could you please take care of shielding whatever is specific for those special runs behind some switch that can be activated via a configuration parameter (in DQM and in the other parts of the  code as well)?

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
> Hi, @perrotta . Should that 2nd PR be added in master only, or also to the backport #41923 ?

Let prepare it for master. The release for next week run sits on a dedicated branch, and therefore protections for the general data taking have a low priority for it

**Comment by grzanka** [comment]
> > Hi, @perrotta . Should that 2nd PR be added in master only, or also to the backport #41923 ?
> 
> Let prepare it for master. The release for next week run sits on a dedicated branch, and therefore protections for the general data taking have a low priority for it

@perrotta I has some troubles to get good connection during ORP meeting yesterday. 

As for DQM guard for T2 plots: we already have a protection mechanism implemented.
If DQM runs in a standard mode then the nT2 DQM producer is not added to the DQM live config.

Only if we switch the PPS DQM to "calibration mode", the T2 DQM producer is activated.

This effectively works as a "guarding flag"

Take a look here:
https://github.com/CTPPS/cmssw/blob/0a38b0798473b67bf0e3e09296a128103c9ba8c7/DQM/Integration/python/clients/ctpps_dqm_sourceclient-live_cfg.py#L94

We are considering further developments of fine control of DQM and reconstruction tasks for T2 at a later stage. 
Anyway we will need to deal with retrieving T2 geometry from the CondDB record and adding another reconstruction steps (to be used for later rereco of collected data). At that place we could have a fine control of T2 reconstruction and DQM options.

**Pull request merged**